### PR TITLE
Prefixed "Spatial" to all hand written components

### PR DESCRIPTION
**Pull request by Vatyx**
Reason for the change is that the `Metadata` component was clashing with a symbol found in some Xbox libs.

### CODE REVIEW

**Comment by danielimprobable** [comment]
Do the component classes need to be accessible through blueprint? otherwise I would probably consider wrapping the classes in a namespace rather than renaming them

**Comment by improbable-valentyn** [comment]
I agree with Daniel that we should put them in a namespace (e.g. `SpatialComponents`).

**Comment by improbable-valentyn** [comment]
After discussing this with Dan, I think a better thing to do right now is to prefix the components with `Spatial` (e.g. `SpatialPosition`, `SpatialEntityAcl` etc).
We should gather opinions from the rest of the team about the use of namespaces and then, if there is a decision to use a namespace for anything that isn't a UClass, apply that to everything that applies.

**Comment by Vatyx** [comment]
Ok, will change the PR to reflect that.

File: Source/SpatialGDK/Public/EngineClasses/SpatialActorChannel.h [hunk 1]
```diff
@@ -74,7 +74,7 @@ class SPATIALGDK_API USpatialActorChannel : public UActorChannel
 
 	void RegisterEntityId(const Worker_EntityId& ActorEntityId);
 	bool ReplicateSubobject(UObject* Obj, const FReplicationFlags& RepFlags);
-	virtual bool ReplicateSubobject(UObject* Obj, FOutBunch& Bunch, const FReplicationFlags& RepFlags) override;
+	virtual bool ReplicateSubobject(UObject* Obj, FOutBunch& Bunch, const FReplicationFlags& RepFlags);
```
Comment (m-samiec) [inline Source/SpatialGDK/Public/EngineClasses/SpatialActorChannel.h]: Why the change?

File: Source/SpatialGDK/Public/Schema/Rotation.h [hunk 2]
```diff
@@ -12,19 +12,19 @@
...
 
-	Rotation() = default;
+	SpatialRotation() = default;
 
-	Rotation(float InPitch, float InYaw, float InRoll)
+	SpatialRotation(float InPitch, float InYaw, float InRoll)
 		: Pitch(InPitch), Yaw(InYaw), Roll(InRoll) {}
 
-	Rotation(const FRotator& Rotator)
+	SpatialRotation(const FRotator& Rotator)
```
Comment (m-samiec) [inline Source/SpatialGDK/Public/Schema/Rotation.h]: Missing include for FRotator

File: Source/SpatialGDK/Public/EngineClasses/SpatialActorChannel.h (hunk 1 above)
Comment (Vatyx) [reply Source/SpatialGDK/Public/EngineClasses/SpatialActorChannel.h]: oh shoot, this shouldn't be here

File: Source/SpatialGDK/Public/Schema/Rotation.h (hunk 2 above)
Comment (Vatyx) [reply Source/SpatialGDK/Public/Schema/Rotation.h]: It's not missing. The other includes have `FRotator` defined.

Comment (m-samiec) [reply Source/SpatialGDK/Public/Schema/Rotation.h]: Via `Platform.h`? Is there another reason to include it? If not I'd replace it with the FRotator include only.

Comment (m-samiec) [reply Source/SpatialGDK/Public/Schema/Rotation.h]: Also realise this is tangential to this PR

Comment (Vatyx) [reply Source/SpatialGDK/Public/Schema/Rotation.h]: Can change.

Comment (Vatyx) [reply Source/SpatialGDK/Public/Schema/Rotation.h]: Yeah I'm glad you realize now several comments in 😡
